File: Infrastructure/Repositories/Dto/EventModels.cs
using System;
using System.Collections.Generic;
using Domain.Events;

namespace Infrastructure.Repositories.Dto
{
    public class EventDto
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public EventType Type { get; set; }
        public string Info { get; set; }
        public string Comment { get; set; }
        public List<Department> Departments { get; set; }
        public List<Volunteer> Volunteers { get; set; }
        public virtual List<Purchase> Purchases { get; set; }
        public virtual List<Employee> Organizers { get; set; }
        public virtual List<Employee> Lecturers { get; set; }

        #region Course or AcademicCompetition
        public Subject Subject { get; set; }
        #endregion

        #region Course
        public decimal? Price { get; set; }
        public int? Duration { get; set; }
        #endregion

        #region SchoolWork
        public string Program { get; set; }
        #endregion

        public List<EventExecutionDto> Executions { get; set; }
    }

     public class EventExecutionDto
    {
        public long Id { get; set; }
        public List<EventDateDto> Dates { get; set; }
        public AddressDto Address { get; set; }

    }

    public class EventDateDto
    {
        public DateTime Date { get; set; }
        public string StartTime { get; set; }
        public string EndTime { get; set; }
    }


}

File: HSEvents.Web/client/models/event.models.ts
import { SubjectModel, Department, Volunteer } from './dictionaries.models';
import { Employee } from './user.models';
import { Address } from './address.models';

export class EventModel {
	id: number;
	name: string;
	type: EventType;
    subject: SubjectModel;
	info: string;
	comment: string;
    duration: number;
	executions: EventExecution[];
    departments: Department[];
    volunteers: Volunteer[];
    lecturers: Employee[];
    organizers: Employee[];
    purchases: Purchase[];
}

export enum EventType {
    Course = 1,
    AcademicCompetition = 2,
    SchoolWork = 3
}

export class EventExecution {
	id: number;
	dates: EventDate[];
	address: Address;
}

export class EventDate {
	id: number;
	date: Date;
	startTime: string;
	endTime: string;
}

export class Purchase {
    id: number;
    name: string;
    price: number;
    description: string;
}